chains/ethereum/server/src/lib.rs

```diff
@@ -38,343 +47,109 @@ impl BlockchainClient for EthereumClient {
     }
 
     async fn new(config: BlockchainConfig, addr: &str) -> Result<Self> {
-        let client = Arc::new(Provider::<Http>::try_from(addr)?);
-        let genesis = client
-            .get_block(0)
-            .await?
-            .context("Failed to get genesis block")?;
-        let genesis_block = BlockIdentifier {
-            index: 0,
-            hash: hex::encode(genesis.hash.as_ref().unwrap()),
-        };
-        Ok(Self {
-            config,
-            client,
-            genesis_block,
-        })
+        GenericEthereumClient::new(config, addr).await
     }
 
     fn config(&self) -> &BlockchainConfig {
-        &self.config
+        match self {
+            GenericEthereumClient::Http(http_client) => http_client.config(),
+            GenericEthereumClient::Ws(ws_client) => ws_client.config(),
+        }
     }
 
     fn genesis_block(&self) -> &BlockIdentifier {
-        &self.genesis_block
+        match self {
+            GenericEthereumClient::Http(http_client) => http_client.genesis_block(),
+            GenericEthereumClient::Ws(ws_client) => ws_client.genesis_block(),
+        }
     }
 
     async fn node_version(&self) -> Result<String> {
-        Ok(self.client.client_version().await?)
+        match self {
+            GenericEthereumClient::Http(http_client) => http_client.node_version().await,
+            GenericEthereumClient::Ws(ws_client) => ws_client.node_version().await,
```

Comment:
why do we need to still support http?

Reply:
I leaved it here just in case we still want to connect to HTTP endpoints, the current implementation supports both http and ws endpoints.

Reply:
But if we want to force WS, I can remove it completly and use WS only.

Reply:
I'm fine with that. No point in keeping code for features we don't need or use